### ISSUE TITLE
block rejected with InvalidKesSignatureOCERT after renewing KES key and opcert.

### ISSUE THREAD

**Issue by jbgi** [input-output-hk/cardano-node#947]
1. Started a three nodes cluster, producing blocks, from genesis:
```
{
    "activeSlotsCoeff": 5.0e-2,
    "protocolParams": {
        "poolDecayRate": 0,
        "poolDeposit": 0,
        "protocolVersion": {
            "minor": 0,
            "major": 0
        },
        "decentralisationParam": 0.99,
        "maxTxSize": 2048,
        "minFeeA": 0,
        "maxBlockBodySize": 2097152,
        "keyMinRefund": 0,
        "minFeeB": 0,
        "eMax": 0,
        "extraEntropy": {
            "tag": "NeutralNonce"
        },
        "maxBlockHeaderSize": 8192,
        "keyDeposit": 0,
        "keyDecayRate": 0,
        "nOpt": 100,
        "rho": 0,
        "poolMinRefund": 0,
        "tau": 0,
        "a0": 0
    },
    "protocolMagicId": 838299499,
    "startTime": "2020-05-12T14:42:00Z",
    "genDelegs": {
        "451bc887ac971729ad290e044b129f43459a50dc9eee584e876e3a08a8b82efe": "90f526b055fede0bc79b45e7348fbbe7a26f211ab80f5f083e2ec1e54e27f989",
        "01770c6f0e6cd19cf271e14b4b84f00975bcc61490c7b447bdc3166244972bb5": "59f52ad68108333a9e435bf76f89c6a1bcbea96221360ccd512be85df1c344a7",
        "eae61f26466397439e36c1c66f9666191618fca92469a1fc5fb076f07254b8f4": "c055bfd8b07fcc11ab16dc3eaba9926c44f3404249e24eaafff3fe2a436141cf"
    },
    "updateQuorum": 12,
    "maxMajorPV": 25446,
    "initialFunds": {
        "82065820260df467bd3415565d35b11ad0a8082258819f570ba4f1d454ddd82dec289420": 30000000000000000,
        "82065820bcbbf562044b7e595bcd8e6fed3bdc1dc21bee241b13c17b586e6d6ed0b686ca": 30000000000000000
    },
    "maxLovelaceSupply": 60000000000000000,
    "networkMagic": 4036000900,
    "epochLength": 432000,
    "staking": null,
    "slotsPerKESPeriod": 60,
    "slotLength": 1,
    "maxKESEvolutions": 14,
    "securityParam": 2160
}
```
keys for the thre nodes were generated with:
```
for i in {1..3}; do
  cardano-cli shelley node key-gen-VRF --verification-key-file node-vrf$i.vkey --signing-key-file node-vrf$i.skey
  cardano-cli shelley node key-gen-KES --verification-key-file node-kes$i.vkey --signing-key-file node-kes$i.skey
  cardano-cli shelley node issue-op-cert --hot-kes-verification-key-file node-kes$i.vkey --cold-signing-key-file ../delegate-keys/delegate$i.skey --operational-certificate-issue-counter ../delegate-keys/delegate-opcert$i.counter --kes-period 0 --out-file node$i.opcert
done
```

2. Then created new KES key and opcert with period 3:
```
 cardano-cli shelley node key-gen-KES \
  --verification-key-file node-kes1.vkey \
  --signing-key-file node-kes1.skey

cardano-cli shelley node issue-op-cert \
  --hot-kes-verification-key-file node-kes1.vkey \
  --cold-signing-key-file ../delegate-keys/delegate1.skey \
  --operational-certificate-issue-counter  ../delegate-keys/delegate-opcert1.counter \
  --kes-period 3 \
  --out-file node1.opcert
```

3. after 3 minutes passed, redeployed the node with the new `node-kes1.skey` and new `node1.opcert`.

4. Seeing in the logs:
```
May 12 14:46:47 a cardano[7199]: fromList [("kind",String "OutcomeTraceForgeEvent"),("difftime",Number 5.297e7),("slot",Number 287.0),("mempoolbytes",Number 0.0),("mempoolnumtx",Number 0.0)]
May 12 14:46:47 a cardano[7199]: Block added to queue: (Point 287, HashHeader {unHashHeader = bf25ef20f762456377dd4d2df1f4aba5e8ab654a6280e30fe553284b9e14a12a}) queue size 1
May 12 14:46:47 a cardano[7199]: Invalid block (Point 287, HashHeader {unHashHeader = bf25ef20f762456377dd4d2df1f4aba5e8ab654a6280e30fe553284b9e14a12a}): ExtValidationErrorHeader (HeaderProtocolError [[OverlayFailure (OcertFailure (InvalidKesSignatureOCERT "Verification failed"))]])
May 12 14:46:47 a cardano[7199]: Valid candidate (Point 268, HashHeader {unHashHeader = 7c7a38f2ee9a1a43713e9b2f4ea8690d9823a32ccace45640f93bd5e59d610c6})
May 12 14:46:47 a cardano[7199]: forgedInvalidSlotLast = 287
May 12 14:46:47 a cardano[7199]: Forged invalid block for slot 287, reason: ValidationError (ExtValidationErrorHeader (HeaderProtocolError [[OverlayFailure (OcertFailure (InvalidKesSignatureOCERT "Verification failed"))]]))
```

**Comment by jbgi** [comment]
After a few minutes the node restart with
```
May 12 15:10:47 a 6y81rnr9wavhi9f8vslp4q7ngcpnzm0d-unit-script-cardano-node-start[10121]: cardano-node: ExceptionInLinkedThread "ThreadId 45" Could not update KES key
May 12 15:10:47 a 6y81rnr9wavhi9f8vslp4q7ngcpnzm0d-unit-script-cardano-node-start[10121]: CallStack (from HasCallStack):
May 12 15:10:47 a 6y81rnr9wavhi9f8vslp4q7ngcpnzm0d-unit-script-cardano-node-start[10121]:   error, called at src/Ouroboros/Consensus/Shelley/Protocol.hs:232:28 in ouroboros-consensus-shelley-0.1.0.0-Gk6Cg7VCxZgLsXtjiyTFKR:Ouroboros.Consensus.Shelley.Protocol
```

**Comment by kevinhammond** [comment]
May be fixed by new releases.

**Comment by jbgi** [comment]
Fixed by #973

**Comment by maciejso** [comment]
What was the reason?
